packages/vite/src/node/server/index.ts

```diff
@@ -489,6 +495,18 @@ export async function createServer(
     handleFileAddUnlink(normalizePath(file), server)
   })
 
+  ws.on('vite:invalidate', async (url: string) => {
+    const mod = moduleGraph.urlToModuleMap.get(url)
+    if (mod && mod.isSelfAccepting && mod.lastHMRTimestamp > 0) {
+      const importers = new Set<ModuleNode>()
+      for (const importer of mod.importers) {
+        importers.add(importer)
+      }
+      const file = getShortName(mod.file!, config.root)
+      updateModules(file, Array.from(importers), mod.lastHMRTimestamp, server)
```

Comment:
This `handleHotUpdate` is not called here.
https://github.com/vitejs/vite/blob/fb6d8cb8233d813a612a98f174db454237598c66/packages/vite/src/node/server/hmr.ts#L97-L102
I understood that handleHotUpdate is already called when selfAccept is done, so there is no need to call it. Am I correct?

Reply:
As I understand it, handleHotUpdate is meant to receive the file that changed and there is no changed file in this case.

Reply:
Ah, that makes sense. Thanks!